practice/rain.py

```diff
@@ -0,0 +1,193 @@
+"""Rain database processor.
+
+Accesses City of Portland rain total database and returns various information
+about rain totals.
+"""
+import urllib.request
+import re
+import os.path
+import statistics
+
+
+def get_daily_totals(filename):
+    """Return a dict of days and daily amounts, as rows from the file."""
+    totals = {}
+    with open(filename) as f:
+        for i in range(11):
+            _ = f.readline()
+        for line in f:
+            day_stats = line.split()[0:2]
+            if day_stats[1] != '-':  # skip entries if there is missing data
+                totals.update({day_stats[0]: int(day_stats[1])})
+    return totals
+
+
+def find_wettest_stats(dates_totals):
+    """Return wettest day (or year) and the amount on the wettest day (or year).
+
+    >>> find_wettest_stats({'30-MAR-2016': 0, '29-MAR-2016': 5})
+    ('29-MAR-2016', 5)
+    """
+    totals_dates = {v: k for k, v in dates_totals.items()}
+    largest = max(totals_dates)
+    return totals_dates[largest], largest
```

Comment:
This works for this very specific case, but the steps encounter some strange behavior that I want you to be aware of so you know to look out for it in the future.

Dictionary keys have to be unique. If a dict comprehension encounters multiple identical keys, only the last one is in the output dict. In this case, yes, there is only one maximum value, but you're not in the most general sense "inverting the dict".
```py
x = {'a': 1, 'b': 1}
inv_x = {v: k for k, v in x.items()
inv_x  #> {1: 'b'}
```

I would suggest you use `max()` with `key`.

```py
def get_second(pair):
    return pair[1]
max(dates_totals.items(), key=get_second)
```

or

```py
from operators import itemgetter
max(dates_totals.items(), key=itemgetter(1))
```

That avoids unpacking and repacking the dictionary and tuples, and I think more directly addresses the operation you want.